src/realm/sync/network/http.hpp

```diff
@@ -306,6 +309,15 @@ struct HTTPParser : protected HTTPParserBase {
         m_socket.async_read_until(m_read_buffer.get(), max_header_line_length, '\n', std::move(handler));
     }
 
+    inline uint32_t hex_to_int(const std::string& hex) throw()
```

Comment:
This `throw()` specification in the function is deprecated `https://en.cppreference.com/w/cpp/language/noexcept_spec`. I think you may want `noexcept(true)` here if you really want to specify that it could throw, or alternatively you could leave it out.

Reply:
I removed this function altogether since `std::strtoul` cleaned up a bunch of things